code/modules/mob/living/living_fov.dm

```diff
@@ -0,0 +1,262 @@
+/// Datumized FOV handler for a mob's client. Handles the visual part of the FOV mechanics.
+/datum/field_of_view
+	/// Reference to the mob owning this
+	var/mob/living/owner
+	/// Currently applied x size of the fov masks
+	var/current_fov_x = BASE_FOV_MASK_X_DIMENSION
+	/// Currently applied y size of the fov masks
+	var/current_fov_y = BASE_FOV_MASK_Y_DIMENSION
+	/// Whether we are applying the masks now
+	var/applied_mask = FALSE
+	/// The angle of the mask we are applying
+	var/fov_angle = FOV_180_DEGREES
+	/// The blocker mask applied to a client's screen
+	var/atom/movable/screen/fov_blocker/blocker_mask
+	/// The shadow mask applied to a client's screen
+	var/atom/movable/screen/fov_shadow/visual_shadow
+	/// Whether our user is alive
+	var/user_living = TRUE
+	/// Whether our user extends his eye view
+	var/user_extends_eye = FALSE
+
+/datum/field_of_view/New(mob/living/our_owner, fov_type = FOV_180_DEGREES)
+	owner = our_owner
+	owner.fov_handler = src
+	var/client/parent_client = owner.client
+	if(!parent_client) //Love client volatility!!
+		qdel(src)
+		return
+
+	fov_angle = fov_type
+	blocker_mask = new
+	visual_shadow = new
+	blocker_mask.dir = owner.dir
+	visual_shadow.dir = owner.dir
+	visual_shadow.alpha = parent_client.prefs.read_preference(/datum/preference/numeric/fov_darkness)
```

Comment:
Will need to check client here too

Reply:
```suggestion
	visual_shadow.alpha = parent_client?.prefs.read_preference(/datum/preference/numeric/fov_darkness)
```

Reply:
Think so? The client was just checked above.

Reply:
It's precisely your comment about clients--from both me and oranges' experience, `/client` can delete whenever, wherever.